### ISSUE TITLE
[Beetle PSX HW][Vulkan] Metal Gear Solid Integral - passing through geometry/out of bounds

### ISSUE THREAD

**Issue by Arthandas** [libretro/beetle-psx-libretro#497]
Sometimes Snake just falls through platforms/stairs to a lower level, goes through walls when leaning against them or goes out of bounds while crawling through vents. So far I played only around 1,5h and it already happened like 6-7 times. It seems pretty random but always happens eventually.

Settings, all default except:
*Video driver changed to Vulcan
*Adaptive smoothing: off
*Internal GPU resolution: 8x
*Internal color depth: 32bpp
*PGXP operation mode: memory only
*PGXP perspective correct texturing: on
*CPU frequency scaling: 150%
*Skip BIOS: on
*Dithering pattern: off
*lottes-multipass_interlaced_glow shader

My specs:
Win10 64x
GTX 1060 6gb (419.35 drivers)
8gb DDR4
i5-7500

**Comment by hizzlekizzle** [comment]
Seems like a PGXP issue. Can you try disabling that and see if it keeps happening?

**Comment by Arthandas** [comment]
I'll play some more and try both memory + cpu and off.

**Comment by rz5** [comment]
Also, if you find a place in-game where you can reliably do one of those shenanigans, consider posting a savestate in here.

**Comment by Arthandas** [comment]
Yes, it's definitely PGXP's fault. With "memory+cpu" I can reliably glitch through keycard door in the attached save state (and presumably all others). "Memory only" is much more random.

Basically equip the keycard and start quickly entering and exiting the room. One of two things will happen:
a) Snake will start passing through the door before its even fully open. If you now unequip the card and try to pass through the door you will glitch into the wall.
b) The door will permanently shut trapping you inside the room. If you keep running or leaning against it (or nearby walls) you should glitch into a wall.

[MGS - Integral (Disc 1) [SLPM 86247].zip](https://github.com/libretro/beetle-psx-libretro/files/3070910/MGS.-.Integral.Disc.1.SLPM.86247.zip)

**Comment by Arthandas** [comment]
After messing with memory + cpu, I set it back to memory only and strangely I had not a single glitch in 4+ hours of play...

**Comment by lmills87** [comment]
I just want to add that I also can glitch out of the map quite easily at the very beginning of the game around the area where the first ration is in the water which PGXP is set to memory+CPU.

**Comment by brad86m** [comment]
Sadly, is an issue with PGXP, and if the Discord chats are anything to go by, the devs of both Duckstation and Beetle have no interest in trying to find and fix the cause for just the few games that have issues. They're also not the main authors of PGXP. Maybe they don't understand how it works, like the original guy does. It hasn't had any updates from the main developer in a very long time.

**Comment by speedfire514** [comment]
Even if it won't be fixed, for the sake of collecting data. Here is how I reproduce that 100% of the time.
In the cell, after meeting Meryl for the first time and defeating waves of Genome soldiers, you are supposed to follow her and use the elevator. Here you can go easily through the left door which will trigger a cutscene and skip an hour of gameplay.
![Bug_PGXP](https://user-images.githubusercontent.com/42750224/152189867-91bfee59-780b-46a1-b2d2-54d9369ec4e4.png)

Another less easy location but at the really beginning of the game, in the first minute of gameplay, you are supposed to evade guards using an elevator, but you have to wait for it to go down. While waiting, you can try to force and fall down the elevator pit. As there is no way to exit that pit, I'm pretty it is supposed to have a collision system/nav mesh preventing it.
